tests/unit/httpd_test.cc

```diff
@@ -908,10 +914,7 @@ SEASTAR_TEST_CASE(test_client_response_eof) {
 
         future<> client = seastar::async([&lcf] {
             auto cln = http::experimental::client(std::make_unique<loopback_http_factory>(lcf));
-            auto req = http::request::make("GET", "test", "/test");
-            BOOST_REQUIRE_EXCEPTION(cln.make_request(std::move(req), [] (const http::reply& rep, input_stream<char>&& in) {
-                return make_exception_future<>(std::runtime_error("Shouldn't happen"));
-            }, http::reply::status_type::ok).get(), std::system_error, [] (auto& ex) {
+            BOOST_REQUIRE_EXCEPTION(make_failing_http_request(cln).get(), std::system_error, [] (auto& ex) {
```

Comment:
I don't understand what I'm seeing here. As far as I know, BOOST_REQUIRE_EXCEPTION already validates that the specific exception type you specified (std::system_error) is the one thrown - and fails the test if the function doesn't throw at all. It looks to me like this whole "shouldn't happen" thing isn't needed at all, and your new function make_failing_http_request shouldn't be needed either.
What am I missing?

Reply:
The make request accepts lambda that returns a future. I do need to return some future from it.
Also note, that BOOST_REQUIRE_EXCEPTION wants system_error with ECONNABORTED code, while "shouldn't happen" exception is of runtime_error type, they are not related at all. The latter, as I wrote, is just because some future should be returned from there.

Reply:
Yes, I know the "shouldn't happen" works because it returns the wrong type of exception so causes the test to fail. But I said that **no exception** would also make the test fail.
I don't know about the future thing and how to make this least ugly. But I still feel that boilerplate code which creates weird exceptions can't be the simplest way to do it.

Reply:
> Yes, I know the "shouldn't happen" works because it returns the wrong type of exception so causes the test to fail. But I said that no exception would also make the test fail.

The test passes. This indirectly proves that this "shouldn't happen" is indeed never called, so it can be assert(false) or return make_ready_future<>() as well.

Reply:
Maybe I'll pass default-initialized noncopyable_function there. If called would just segfault , but since it's not called it would work